components/homme/src/theta-l/share/model_init_mod.F90

```diff
@@ -72,49 +72,33 @@ subroutine model_init2(elem,hybrid,deriv,hvcoord,tl,nets,nete )
       enddo
 
       ! initialize reference states used by hyberviscosity
-      ps_ref(:,:) = hvcoord%ps0 * exp ( -elem(ie)%state%phis(:,:)/(Rgas*TREF)) 
-      do k=1,nlev
-         elem(ie)%derived%dp_ref(:,:,k) = ( hvcoord%hyai(k+1) - hvcoord%hyai(k) )*hvcoord%ps0 + &
-              (hvcoord%hybi(k+1)-hvcoord%hybi(k))*ps_ref(:,:)
-      enddo
-      call set_theta_ref(hvcoord,elem(ie)%derived%dp_ref,elem(ie)%derived%theta_ref)
-      temp=elem(ie)%derived%theta_ref*elem(ie)%derived%dp_ref
-      call phi_from_eos(hvcoord,elem(ie)%state%phis,&
-           temp,elem(ie)%derived%dp_ref,elem(ie)%derived%phi_ref)
-      if (hv_ref_profiles==0) then
-         ! keep PHI profile, but dont use theta and dp:
-         elem(ie)%derived%theta_ref=0
-         elem(ie)%derived%dp_ref=0
-      endif
-      if (hv_ref_profiles==1) then
-         ! keep all profiles
-      endif
-      if (hv_ref_profiles==2) then
-         ! keep only dp_ref
-         elem(ie)%derived%theta_ref=0
-      endif
+      call initialize_reference_states(hvcoord, elem(ie)%state%phis, &
+                                       elem(ie)%derived%dp_ref,      &
+                                       elem(ie)%derived%theta_ref,   &
+                                       elem(ie)%derived%phi_ref)
 
       if (hv_theta_correction/=0) then
```

Comment:
minor question -- why not include this code in the routine above?

Reply:
wouldn't this be needed in unit tests if related code in biharmonic_wk_theta routine is on?

Reply:
the code related to "hv_theta_correction" is unrelated to this PR (and not coded in EAMxx).  It should be removed or tested (on the Fortran side), but lets not delay this PR for that.

Reply:
The above is computing just the reference profiles, where below is computing `lap_p_wk`. Are these related?